src/utils.js

```diff
@@ -5,3 +5,45 @@ export const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayo
 // useForceUpdate hook
 const forcedReducer = state => state + 1;
 export const useForceUpdate = () => useReducer(forcedReducer, 0)[1];
+
+export const createDeepProxy = (obj, affected) => {
+  const proxyMap = {};
+  const handler = {
+    get: (target, key) => {
+      if (!affected.has(target)) {
+        affected.set(target, [key]);
+      } else {
+        const used = affected.get(target);
+        if (!used.includes(key)) used.push(key);
```

Comment:
🚀

Reply:
This turns out to be "really" slow when `used` is big.

Reply:
Have you tried `Set`?

Probably it's time to use some V8-level hacks, like a using different approaches and data structures depending on "size".

Reply:
Yes, I used `Set`. 8373b9721ef74304d6610d5927a7bb680688c738